dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/factory.py

```diff
@@ -2616,26 +2621,119 @@ def get_list_for_evalcalculatedvari(self):
         return self.list_for_evalcalculatedvari
 
     ##
-    # prepare the lines that constitues the body for evalJCalculatedVarI
+    # prepare the lines that constitues the body for evalCalculatedVarIAdept
     # @param self : object pointer
     # @return
-    def prepare_for_evaljcalculatedvar(self):
-        self.list_for_evaljcalculatedvari.append("  // not needed\n")
+    def prepare_for_evalcalculatedvariadept(self):
+        trans = Transpose(self.reader.auxiliary_vars_to_address_map, self.reader.derivative_residual_vars + self.reader.assign_residual_vars)
+        ptrn_vars = re.compile(r'x\[(?P<varId>[0-9]+)\]')
+        self.list_for_evalcalculatedvariadept.append("  data->simulationInfo->discreteCall = 1;\n")
```

Comment:
Same remark on the discreteCall value. Not very sure of the behavior

Reply:
Solved with same solution